lib/logger.js

```diff
@@ -70,20 +73,35 @@ function now () {
 }
 
 function createLogger (options) {
-  if (!options.logger) {
+  // if no logger is provided, then create a default logger
+  if (!options.loggerInstance && !options.logger) {
     const logger = nullLogger
     logger.child = () => logger
     return { logger, hasLogger: false }
   }
 
-  if (validateLogger(options.logger)) {
+  if (options.logger && options.loggerInstance) {
+    throw new FST_ERR_LOGGER_AND_LOGGER_INSTANCE_PROVIDED()
+  }
+
+  // check if the logger instance has all required properties
+  if (validateLogger(options.loggerInstance)) {
     const logger = createPinoLogger({
-      logger: options.logger,
-      serializers: Object.assign({}, serializers, options.logger.serializers)
+      logger: options.loggerInstance,
+      serializers: Object.assign({}, serializers, options.loggerInstance.serializers)
     })
     return { logger, hasLogger: true }
   }
 
+  // if a logger instance is passed to logger, throw an exception
+  if (validateLogger(options.logger)) {
+    throw FST_ERR_LOG_INVALID_LOGGER_CONFIG()
+  }
+
+  if (options.loggerInstance) {
```

Comment:
and here we dont use validateLogger to check if the logger has the right shape. i think, the check should be in line 88. first check if loggerInstance is set, if so check with validateLogger if it is a valid instance and do the instantiation and if not throw the Error.

Reply:
Line 101 basically says that if a logger fn isn't passed to `loggerInstance`, but `loggerInstance` still contains some value, we make an assumption that a pino config or some config was passed to `loggerInstance` even though it was supposed to be passed to `logger`. Hence, we throw an exception saying `loggerInstance only accepts a logger not a configuration object.`

Reply:
Let me repeat what you basically wrote: Something was passed but an loggerInstance. So the Error can not say the second half sentence "not a configuration object".

Reply:
Ah gotcha. What do you think of `A valid logger function was not passed to loggerInstance`?

Reply:
@aarontravass could you breaking into two funcs? `validateLoggerInstance` and `validateLoggerConfig` would be clearer and if needed validateLoggerConfig... also spot checked pino there are some errors nested in a tool for normalizing the args/options... not sure if thats best interface for a dev though (I'm assuming devs know its pino but 🤷)

https://github.com/pinojs/pino/blob/master/lib/tools.js#L286